Frontend/ui-customer/src/Flow.purs

```diff
@@ -177,11 +179,38 @@ baseAppFlow (GlobalPayload gPayload) refreshFlow = do
               case triggerSignatureOtpResp.token of
                 Just token -> setValueToLocalStore REGISTERATION_TOKEN token
                 Nothing -> pure unit
-              currentFlowStatus
-            Left err -> do
-              liftFlowBT $ pure $ runFn3 emitJOSEvent "java" "onEvent" $ encode $  EventPayload { event : "signature_auth_failed", payload : Nothing}
-              pure unit
-        Nothing -> if (showCarouselScreen FunctionCall) then welcomeScreenFlow else enterMobileNumberScreenFlow
+              handleDeepLinks (GlobalPayload merchantPayload) false
+        Left err -> do
+          liftFlowBT $ pure $ runFn3 emitJOSEvent "java" "onEvent" $ encode $  EventPayload { event : "signature_auth_failed", payload : Nothing}
+          pure unit
+    Nothing -> if getValueToLocalStore REGISTERATION_TOKEN /= "__failed" && getValueToLocalStore REGISTERATION_TOKEN /= "(null)" &&  (isNothing $ payload.signatureAuthData)
+    then handleDeepLinks (GlobalPayload merchantPayload) false
+    else if (showCarouselScreen FunctionCall) then welcomeScreenFlow else enterMobileNumberScreenFlow
+
+handleDeepLinks :: GlobalPayload -> Boolean -> FlowBT String Unit
+handleDeepLinks (GlobalPayload merchantPayload) shouldSkipDefault = do
```

Comment:
what is `shouldSkipDefault` used for?

Reply:
this is to handle deeplink and skip currentFlowStatus if is called from currentFlowStatus for user who are not logged in.

Reply:
let's rename to `skipDefaultCase`